### PR TITLE
add s390 buildroot and qemu commands

### PR DESCRIPTION
**Pull request by nickdesaulniers**
The SystemZ maintainer in LLVM, Ulrich Weigand, sugguested that
buildroot should work if manually switched to use glibc.

Add a buildroot config, and qemu command to boot.

Currently require's David Hildenbrand <david@redhat.com>'s QEMU patches
at
https://lists.nongnu.org/archive/html/qemu-s390x/2021-01/msg00035.html
to boot.

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
cc @davidhildenbrand

**Review by nathanchance — APPROVED** [review]
This looks good to me.

For others wanting to test this, grab [`b4`](https://git.kernel.org/pub/scm/utils/b4/b4.git/about/) and run the following commands (assuming that `ninja`, `s390x-linux-gnu` tools, and `clang` are in your `PATH`).

Building QEMU:

```
$ git clone --recursive https://git.qemu.org/git/qemu.git

$ cd qemu

$ b4 am -o - https://lore.kernel.org/qemu-devel/20210108132049.8501-1-david@redhat.com/ | git apply -3v

$ mkdir build

$ cd build

$ ../configure --target-list=s390x-softmmu

$ ninja
```

Building Linux:

```
$ cd ../..

$ git clone https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/

$ cd linux

$ make -skj"$(nproc)" ARCH=s390 CC=clang CROSS_COMPILE=s390x-linux-gnu- O=out/s390 distclean defconfig bzImage
```

Testing it all with this pull:

```
$ cd ..

$ git clone -b buildroot_2020_11 https://github.com/ClangBuiltLinux/boot-utils

$ PATH=${PWD}/qemu/build:${PATH} boot-utils/boot-qemu.sh -a s390 -k linux/out/s390 -t 45s
```

**Comment by davidhildenbrand** [comment]
QEMU patches were reviewed, so they should land in qemu/master in a 1-3 weeks.

File: boot-qemu.sh [hunk 1]
```diff
@@ -203,6 +203,12 @@ function setup_qemu_args() {
             QEMU=(qemu-system-riscv64)
             ;;
 
+        s390)
+            KIMAGE=bzImage
+            QEMU_ARCH_ARGS=(-M s390-ccw-virtio)
+            QEMU=(qemu-system-s390x)
```
Comment (davidhildenbrand) [inline boot-qemu.sh:209]: Using "-smp", as done on x86-64 AFAIKs might reduce boot times further. SMP under QEMU/TCG with MTTCG (one thread per vCPU, usually the default) has been working reliably for some years now.

**Pull request merged**